test/test_tensordict.py

```diff
@@ -1137,14 +1176,10 @@ def test_update(self, td_name, device, clone):
             clone=clone,
         )
         keys = keys.union({("newnested", "x"), ("newnested", "w")})
-        assert keys == set(td.keys(True))
+        assert keys == set(list(keys_view))
         td.update({("newnested",): {"v": torch.zeros(td.shape)}}, clone=clone)
-        keys = keys.union(
-            {
-                ("newnested", "v"),
-            }
-        )
-        assert keys == set(td.keys(True))
+        keys = keys.union({("newnested", "v"),})
+        assert keys == set(list(keys_view))
```

Comment:
(deleting the comma will stop the formatter exploding onto multiple lines)

```suggestion
        keys = keys.union({("newnested", "v")})
        assert keys == set(keys_view)
```